src/pages/SearchList/SearchList.js

```diff
@@ -1,8 +1,293 @@
-import React from "react";
+import React, { useState } from "react";
 import styled from "styled-components";
+import * as s from "./Styled.SearchList";
+import CalendarModal from "./Calendar";
+import LocationModal from "./LocationModal";
+import HeadcountModal from "./HeadcountModal";
+import { faLocationDot } from "@fortawesome/free-solid-svg-icons";
+import { faCalendar } from "@fortawesome/free-solid-svg-icons";
+import { faUserLarge } from "@fortawesome/free-solid-svg-icons";
 
 const Searchlist = () => {
-  return <div>SearchList</div>;
+  const [calendar, setCalander] = useState(false);
+  const [location, setLocation] = useState(false);
+  const [headcount, setHeadcount] = useState(false);
+  const [adult, setAdult] = useState(1);
+  const [child, setChild] = useState(0);
+  const [range, setRange] = useState(100);
+  const [rating, setRating] = useState(100); // initial rating value
+  const [locationInput, setLocationInput] = useState("");
+
+  // Catch Rating value
+
+  const openCalendar = () => {
+    setCalander(!calendar);
+  };
+  const openLocation = () => {
+    setLocation(!location);
+  };
+  const openHeadcount = () => {
+    setHeadcount(!headcount);
+  };
+
+  const handleRating = (rate: number) => {
+    setRating(rate);
+    // other logic
+  };
+
+  return (
+    <div>
+      <Nav />
+      <s.SearchContainer>
+        <s.SearchInnerContainer>
+          <s.SearchLocationContainer>
+            <s.LocationInputBox>
+              <s.StyledFontAwesomeIcon icon={faLocationDot} size="2x" />
+              <s.LocationInput
+                onChange={e => {
+                  setLocationInput(e.target.value);
+                }}
+                onFocus={openLocation}
+              />
+            </s.LocationInputBox>
+            {location ? <LocationModal locationInput={locationInput} /> : null}
```

Comment:
삼항 연산자 대신 논리곱 연산자를 활용한 단축평가를 사용해서 조건부 렌더링하시면 될 것 같습니다.
아래 리뷰하지 않은 다른 부분들도 동일하게 수정해주세요!
```suggestion
            {location && <LocationModal locationInput={locationInput} />}
```

Reply:
수정했습니다